aurora-architecture-overview.md

```diff
@@ -0,0 +1,249 @@
+# Aurora Dashboard – Architecture Overview
+
+## Introduction
+
+Aurora Dashboard is a modern replacement for the legacy OpenStack Horizon interface. It is designed to be fast, modular, and developer-friendly, offering a clean and type-safe experience across the stack. By following a Backend-for-Frontend (BFF) architecture, we abstract away OpenStack complexity and deliver a UI that is decoupled, composable, and scalable.
+
+This document describes the architecture, conventions, technologies, and implementation strategies used to build and maintain the Aurora Dashboard.
+
+---
+
+## Tech Stack and Tooling
+
+| Area              | Tooling                             |
+| ----------------- | ----------------------------------- |
+| UI Framework      | React 19, juno                      |
+| Styling           | Tailwind CSS v4                     |
+| Routing           | TanStack Router (v1)                |
+| API Communication | tRPC + Zod                          |
+| Server Runtime    | Fastify                             |
+| i18n              | Lingui                              |
+| Dev Tools         | Vite, Vitest, pnpm, TurboRepo       |
+| Code Quality      | ESLint, Husky, Conventional Commits |
+
+---
+
+## Monorepo Structure
+
+Aurora uses a pnpm-based monorepo managed by TurboRepo. Shared tooling is centralized, and apps and packages are clearly separated.
+
+```
+/aurora
+  /apps
+    /aurora-portal         # Main frontend application
+  /packages
+    /aurora-sdk            # tRPC client wrapper
+    /signal-openstack      # OpenStack integration layer
+    /config                # Shared tsconfig, eslint, tailwind config
+    /template              # Internal package scaffold template
+```
+
+---
+
+## Frontend Routing and State – TanStack Router
+
+We use TanStack Router v1 with file-based routing to build our UI declaratively and with type safety at every level.
+
+### File-based Routing Conventions
+
+To maximize DX and enable type-safe navigation:
+
+- All route files for dynamic segments use a $ prefix, e.g. /$projectId.tsx
+- UI subfolders that shouldn’t become routes (e.g., components) are prefixed with -, e.g. `-components/`
+- Each route export `component`, `loader`, and optionally `beforeLoad`
+- Internal folders like are excluded from route generation
```

Comment:
There seems to be something missing: 
"Internal folders like [MISSING] are excluded..."